src/goto-checker/stop_on_fail_verifier.h

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************\
+
+Module: Goto Verifier for stopping at the first failing property
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Goto Verifier for stopping at the first failing property
+
+#ifndef CPROVER_GOTO_CHECKER_STOP_ON_FAIL_VERIFIER_H
+#define CPROVER_GOTO_CHECKER_STOP_ON_FAIL_VERIFIER_H
+
+#include "bmc_util.h"
+#include "goto_verifier.h"
+
+/// Stops when the first failing property is found.
+/// Requires an incremental goto checker that is a
+/// `goto_trace_providert` and `witness_providert`.
+template <class incremental_goto_checkerT>
+class stop_on_fail_verifiert : public goto_verifiert
```

Comment:
I am wondering whether a `stop_on_fail_verifiert` genuinely _has_ an `incremental_goto_checkerT` or rather should _be_ an `incremental_goto_checkerT`?!

Reply:
You'd still need a wrapper around the `incremental_goto_checkert` to do the result reporting.